In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
dataset = pd.read_csv("Market_Basket_Optimisation.csv" , header=None)
dataset.head()

0          1           2                 3             4   \
0         shrimp    almonds     avocado    vegetables mix  green grapes   
1        burgers  meatballs        eggs               NaN           NaN   
2        chutney        NaN         NaN               NaN           NaN   
3         turkey    avocado         NaN               NaN           NaN   
4  mineral water       milk  energy bar  whole wheat rice     green tea   

                 5     6               7             8             9   \
0  whole weat flour  yams  cottage cheese  energy drink  tomato juice   
1               NaN   NaN             NaN           NaN           NaN   
2               NaN   NaN             NaN           NaN           NaN   
3               NaN   NaN             NaN           NaN           NaN   
4               NaN   NaN             NaN           NaN           NaN   

               10         11     12     13             14      15  \
0  low fat yogurt  green tea  honey  salad  mineral water  salmon   
1             NaN        NaN    NaN    NaN            NaN     NaN   
2             NaN        NaN    NaN    NaN            NaN     NaN   
3             NaN        NaN    NaN    NaN            NaN     NaN   
4             NaN        NaN    NaN    NaN            NaN     NaN   

                  16               17       18         19  
0  antioxydant juice  frozen smoothie  spinach  olive oil  
1                NaN              NaN      NaN        NaN  
2                NaN              NaN      NaN        NaN  
3                NaN              NaN      NaN        NaN  
4                NaN              NaN      NaN        NaN

In [3]:
#We need list of list for our apriori implementation
transcations = []

In [4]:
for i in range(len(dataset)):
    transcations.append([str(dataset.values[i,j]) for j in range(0,20)])

transcations[0]

['shrimp',
 'almonds',
 'avocado',
 'vegetables mix',
 'green grapes',
 'whole weat flour',
 'yams',
 'cottage cheese',
 'energy drink',
 'tomato juice',
 'low fat yogurt',
 'green tea',
 'honey',
 'salad',
 'mineral water',
 'salmon',
 'antioxydant juice',
 'frozen smoothie',
 'spinach',
 'olive oil']

In [5]:
len(transcations)

7501

    Support = (3x7) / 7501 i.e product buyed at least 3 times 
    Confidence = 0.20 i.e 20% 
    lift = 3
    min_length = 2 i.e atleast 2 different products

In [6]:
from apyori import apriori
rules = apriori(transcations,min_support= 0.003,min_confidence= 0.2,min_lift= 3, min_length = 2)

#### Visualizing Rules now.

In [7]:
results = list(rules)

In [8]:
clean_results = []
for i in range(0, len(results)):
    clean_results.append('RULE: ' + str(results[i][0]) + '   SUPPORT:' + str(results[i][1]) + '   CONFIDENCE:' + str(results[i][2][0][2]) + '   LIFT:' + str(results[i][2][0][3]))

In [13]:
clean_results[4]

"RULE: frozenset({'herb & pepper', 'ground beef'})   SUPPORT:0.015997866951073192   CONFIDENCE:0.3234501347708895   LIFT:3.2919938411349285"